src/python/pants/backend/java/compile/javac.py

```diff
@@ -42,16 +44,25 @@ class CompileJavaSourceRequest(ClasspathEntryRequest):
     field_sets = (JavaFieldSet, JavaGeneratorFieldSet)
 
 
+@dataclass(frozen=True)
+class JavacRequest:
+    jdk: JdkEnvironment
+    direct_dependencies: tuple[ClasspathEntry, ...]
+    input_digest: Digest
+    input_filenames: tuple[str, ...]
+    output_filename: str
+    exports: tuple[ClasspathEntry, ...]
+    resolve: CoursierResolveKey  # Can we get rid of this
```

Comment:
Unless I'm missing something, I don't see it in the rule for `JavacRequest`? 

Generally I think this should not be a part of a generic `JavacRequest` - the idea of this type is that it has zero knowledge about Pants-isms like Targets, Options, Resolves.

Reply:
+1